peer/pendingheap/list_test.go

```diff
@@ -675,3 +685,44 @@ func TestPeerHeapList(t *testing.T) {
 var noShuffle ListOption = func(c *listConfig) {
 	c.shuffle = false
 }
+
+func TestFailFastConfig(t *testing.T) {
+	conn, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	require.NoError(t, conn.Close())
+
+	serviceName := "test"
+	config := whitespace.Expand(fmt.Sprintf(`
+		outbounds:
+			nowhere:
+				http:
+					fewest-pending-requests:
+						peers:
+							- %q
+						capacity: 10
+						failFast: true
+	`, conn.Addr()))
+	cfgr := yarpcconfig.New()
+	cfgr.MustRegisterTransport(http.TransportSpec())
```

Comment:
Can we use the [`FakeTransportSpec](https://sourcegraph.com/github.com/yarpc/yarpc-go/-/blob/yarpctest/fake_config.go#L70)` here instead of depending on the HTTP package?

Reply:
I tried the option first. The wiring is a bit too hard.